src/autoscaler/db/sqldb/scheduler_sqldb.go

```diff
@@ -82,3 +83,23 @@ func (sdb *SchedulerSQLDB) GetActiveSchedules() (map[string]*models.ActiveSchedu
 	return schedules, nil
 
 }
+
+func (sdb *SchedulerSQLDB) SynchronizeActiveSchedules(appIdMap map[string]bool) error {
+	if len(appIdMap) == 0 {
+		sdb.logger.Debug("No application exists")
+		return nil
+	}
+	var query string = "DELETE FROM app_scaling_active_schedule WHERE app_id NOT IN("
```

Comment:
Just adding to the NOT IN discussion, if the input to NOT IN gets too large, the query might fail.